File: Strings/Caesar Cipher Encryptor/Semordnilap/Sol.py
# O(n*m) time and space
# n - number of words
# m - length of longest word in word

def semordnilap(words):
    # Write your code here.
    wordsSet=set(words)
    ret=[]
    for word in words:
        reversed=word[::-1]
        if reversed in wordsSet and reversed != word:
            ret.append([word,reversed])
            wordsSet.remove(word)
            wordsSet.remove(reversed)
    return ret

File: Strings/Caesar Cipher Encryptor/Semordnilap/Sol.js
// O(n*m) time and space
// n - number of words
// m - length of longest word in word

function semordnilap(words) {
  // Write your code here.
  const wordsSet = new Set(words);
  const semordnilaps = [];

  for (let word of words) {
    const reversed = word.split("").reverse().join("");
    console.log(word, reversed, wordsSet, wordsSet.has(reversed));
    if (wordsSet.has(reversed)) {
      console.log(word, wordsSet);
      semordnilaps.push([word, reversed]);
      wordsSet.delete(word);
      wordsSet.delete(reversed);
    }
  }
  return semordnilaps;
}
